src/pages/my-account/users/index.tsx

```diff
@@ -0,0 +1,225 @@
+    return (
+      <Button
+        variant="primary"
+        disabled={isUserFormOpen}
+        onClick={() => setUserFormOpen(true)}
+        disableElevation
+        id="formOpenButton"
+      >
+        <span style={{ display: 'flex', alignItems: 'center' }}>
+          <AddCircleOutlineIcon style={{ marginRight: '8px', width: '19px' }} />
+          <span style={{ paddingTop: '2px', fontWeight: '400' }}>{t('add-user')}</span>
+        </span>
+      </Button>
+    )
+  }
+
+  return (
+    <Grid>
+      <Grid item style={{ marginTop: '10px', marginBottom: '40px' }}>
+        <BackButtonLink aria-label={t('my-account')} href="/my-account">
+          <ChevronLeftIcon />
+          {mdScreen && <Typography variant="body1">{t('my-account')}</Typography>}
+        </BackButtonLink>
+        <Box sx={style.heading}>
+          <Typography
+            variant={mdScreen ? 'h1' : 'h2'}
+            sx={{ textAlign: { xs: 'center' }, marginTop: { xs: '-3.6rem', md: 0 } }}
+          >
+            {t('users')}
+          </Typography>
+        </Box>
+        <Grid container>
+          <Grid item xs={12} md={12}>
+            {mdScreen && <AddUserButton />}
+            {!mdScreen && !isUserFormOpen && <AddUserButton />}
+            {isUserFormOpen && (
+              <UserForm isEditMode={false} closeUserForm={() => setUserFormOpen(false)} />
+            )}
+          </Grid>
+        </Grid>
+      </Grid>
+      <Grid item>
+        <SearchBoxContainer>
+          <SearchBar
+            onSearch={(searchText) =>
+              setPaginationState({
+                ...paginationState,
+                searchTerm: searchText,
+              })
+            }
+            placeHolder={t('user-search-placeholder')}
+            searchTerm={paginationState.searchTerm}
+            showClearButton={true}
+          />
+        </SearchBoxContainer>
+
+        {isLoading ? (
+          <Box style={{ width: '100%', display: 'flex', justifyContent: 'center' }}>
+            <CircularProgress />
+          </Box>
+        ) : (
+          <>
+            <UserTable
+              b2bAccountUsers={data?.items}
+              deleteUser={confirmDelete}
+              editUserId={editUserId}
+              setEditUserId={(id: Maybe<string> | undefined) => setEditUserId(id)}
+            />
+            <PaginationContainer>
+              <Pagination
+                count={data?.pageCount || 0}
+                shape={`rounded`}
+                onChange={(event: ChangeEvent<any>, page: number) =>
+                  setPaginationState({
+                    ...paginationState,
+                    startIndex: (data?.pageSize || 0) * (page - 1),
+                  })
+                }
+                size="small"
+              />
+              <Typography sx={style.perPageItemText}>{getPerPageItemText()}</Typography>
+            </PaginationContainer>
+          </>
+        )}
+      </Grid>
+    </Grid>
+  )
+}
+
+export default UsersPage
```

Comment:
**1. Naming conventions are inconstant**
`[isUserFormOpen, setUserFormOpen] = useState(false)`
should be 
`[isUserFormOpen, setIsUserFormOpen] = useState<boolean>(false)`

`<UserForm isEditMode={false} closeUserForm={() => setUserFormOpen(false)} />`
should be 
`<UserForm isEditMode={false} onClose={() => setIsUserFormOpen(false)} />`

```
<UserTable
              b2bAccountUsers={data?.items}
              deleteUser={confirmDelete}
              editUserId={editUserId}
              setEditUserId={(id: Maybe<string> | undefined) => setEditUserId(id)}
            />
```
		
should be 
```
<UserTable
              b2bAccountUsers={data?.items}
              onDelete={handleDelete}      
              onEdit={handleEdit}	/>
```
			  

**2. Apply correct types (no need of Maybe)**
`const [editUserId, setEditUserId] = useState<Maybe<string> | undefined>(undefined)`

**3. Unnecessary property, page: publicRuntimeConfig.b2bUserListing.defaultPage,**
In 
 ```
const [paginationState, setPaginationState] = useState({
    searchTerm: '',
    page: publicRuntimeConfig.b2bUserListing.defaultPage,
    pageSize: publicRuntimeConfig.b2bUserListing.defaultPageSize,
    startIndex: publicRuntimeConfig.b2bUserListing.defaultStartIndex,
  })
```


**4.  Unnecessary properties**
```
const { removeCustomerB2bUser } = useRemoveCustomerB2bUserMutation({
    removeCustomerB2bAccountUser: true,
    delay: 1000,
  })
```
 
  Why removeCustomerB2bAccountUser: true parameter?
  Why delay: 1000?


**5.  Make function reusable (put in lib/helper)**
  const getPerPageItemText = () => {


**6. Use translation**    
if (!data) return `${mdScreen && t('displaying')} 0 - 0 of 0`
	`${mdScreen && t('displaying')} ${startRange} - ${
	 primaryButtonText: 'Delete',

 
**7. Unnecessary props**
`<UserTable editUserId={editUserId} />`
 No need to pass editUserId={editUserId}, <UserTable/> should handle it internally

**8. Separate the logic and JSX, ex. create separate function and use it.**
```<Pagination
                count={data?.pageCount || 0}
                shape={`rounded`}
                onChange={(event: ChangeEvent<any>, page: number) =>
```	
 should be 	 `<Pagination onChange={handleChange}' ...`
   
```
 <SearchBar
            onSearch={(searchText) =>
              setPaginationState({
                ...paginationState,
                searchTerm: searchText,
              })
            }
```

 should be 	 `<SearchBar onSearch={handleSearch}' ...`

Reply:
Done